packages/truffle-compile/compilerSupplier.js

```diff
@@ -286,18 +315,32 @@ CompilerSupplier.prototype.getBuilt = function(buildType){
  * @param  {String}  localPath
  * @return {Boolean}
  */
-CompilerSupplier.prototype.isLocal = function(localPath){
+CompilerSupplier.prototype.isLocal = function(localPath) {
   return fs.existsSync(localPath) || path.isAbsolute(localPath);
 }
 
+/**
+ * Returns a valid version name if compiler file is cached
+ * @param  {String}  localPath
+ * @return {Boolean}
+ */
+CompilerSupplier.prototype.versionIsCached = function(version) {
+  const cachedCompilerFileNames = fs.readdirSync(this.cachePath);
+  const cachedVersions = cachedCompilerFileNames.map((fileName) => {
+    const match = fileName.match(/-v(.*)\+/);
```

Comment:
Is this a reliable regex to use to parse out version numbers from the compiler names?